ts/indicators/aroon.src.ts

```diff
@@ -0,0 +1,230 @@
+        i;
+
+    for (i = 1; i < arr.length; i++) {
+        if (
+            extreme === 'max' && arr[i] >= extremeValue ||
+            extreme === 'min' && arr[i] <= extremeValue
+        ) {
+            extremeValue = arr[i];
+            valueIndex = i;
+        }
+    }
+
+    return valueIndex;
+}
+/* eslint-enable valid-jsdoc */
+
+/**
+ * The Aroon series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.aroon
+ *
+ * @augments Highcharts.Series
+ */
+H.seriesType<Highcharts.AroonIndicator>(
+    'aroon',
+    'sma',
+    /**
+     * Aroon. This series requires the `linkedTo` option to be
+     * set and should be loaded after the `stock/indicators/indicators.js`.
+     *
+     * @sample {highstock} stock/indicators/aroon
+     *         Aroon
+     *
+     * @extends      plotOptions.sma
+     * @since        7.0.0
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, compare, compareBase, joinBy, keys,
+     *               navigatorOptions, pointInterval, pointIntervalUnit,
+     *               pointPlacement, pointRange, pointStart, showInNavigator,
+     *               stacking
+     * @optionparent plotOptions.aroon
+     */
+    {
+        /**
+         * Paramters used in calculation of aroon series points.
+         *
+         * @excluding periods, index
+         */
+        params: {
+            /**
+             * Period for Aroon indicator
+             */
+            period: 25
+        },
+        marker: {
+            enabled: false
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>Aroon Up: {point.y}<br/>Aroon Down: {point.aroonDown}<br/>'
+        },
+        /**
+         * aroonDown line options.
+         */
+        aroonDown: {
+            /**
+             * Styles for an aroonDown line.
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * [plotOptions.aroon.color](#plotOptions.aroon.color).
+                 *
+                 * @type {Highcharts.ColorString}
+                 */
+                lineColor: undefined
+            }
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    H.merge(multipleLinesMixin, {
+        nameBase: 'Aroon',
+        pointArrayMap: ['y', 'aroonDown'],
+        pointValKey: 'y',
+        linesApiNames: ['aroonDown'],
+        getValues: function (
+            series: Highcharts.AroonIndicator,
+            params: Highcharts.AroonIndicatorParamsOptions
+        ): Highcharts.IndicatorMultipleValuesObject {
+            var period: number = params.period,
+                xVal: (Array<number>|undefined) = series.xData,
```

Comment:
You should remove `undefined` here and cast the right side to `any`, because it simply needs a default value. And after that you can remove all the `xVal as any` in the code below.